### PR TITLE
[FrameworkBundle] Calls support for debug:container

### PR DESCRIPTION
**Pull request by JhonnyL**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | no |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |
- Show methods to be called after service initialization in `debug:container` command

### CODE REVIEW

**Comment by iltar** [comment]
What does it look like with multiple calls?

**Comment by JhonnyL** [comment]
It will look like this.

screen shot 2016-04-03 at 12 20 47

**Comment by javiereguiluz** [comment]
I think this would be a nice addition :+1:  Thanks @JhonnyL

**Comment by iltar** [comment]
Would it be possible to show arguments of the setters as well? I think that
would be an even greater addition

**Comment by JhonnyL** [comment]
It is possible. Still the command doesn't even show arguments for the debugged service itself. I would hope we can have this merged first and then save arguments support for service and calls etc. to another PR.

We would also need to consider if it's worth it e.g. private services as arguments will make output look really ugly and bloated.

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php [hunk 1]
```diff
@@ -214,6 +214,13 @@ protected function describeContainerDefinition(Definition $definition, array $op
             }
         }
 
+        $calls = $definition->getMethodCalls();
+        if (count($calls) > 0) {
+            foreach ($calls as $callData) {
+                $output .= "\n".'- Call: `'.$callData[0].'`';
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php]: In markdown, should be instead use a sublist ?

```markdown
- Calls:
    - `setParameter`
    - `getParameter`
```

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php [hunk 2]
```diff
@@ -214,6 +214,13 @@ protected function describeContainerDefinition(Definition $definition, array $op
             }
         }
 
+        $calls = $definition->getMethodCalls();
+        if (count($calls) > 0) {
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php]: for now, this check is useless, as you only iterate over the empty array

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php (hunk 1 above)
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php]: OK, the current proposal is consistent with the display of tags, so this is fine

**Comment by JhonnyL** [comment]
Updated PR. Removed useless count check.

**Comment by stof** [comment]
:+1:
Status: reviewed

**Comment by aitboudad** [comment]
:+1:

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php [hunk 3]
```diff
@@ -214,6 +214,11 @@ protected function describeContainerDefinition(Definition $definition, array $op
             }
         }
 
+        $calls = $definition->getMethodCalls();
+        foreach ($calls as $callData) {
+            $output .= "\n".'- Call: `'.$callData[0].'`';
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php]: Imo this should be "Calls" do be consistent with the other descriptors.

Comment (aitboudad) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php]: I'm not sure we already using the same thing for [Tag](https://github.com/JhonnyL/symfony/blob/calls-support-for-debug-container/src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php#L225) and [Autowiring Type](https://github.com/JhonnyL/symfony/blob/calls-support-for-debug-container/src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php#L194) and if we use `Calls` we should list them in the same row, did you see how it [looks](https://cloud.githubusercontent.com/assets/1104667/14232141/d09982ac-f99e-11e5-9432-4d97eb7db05b.png) ?

Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php]: Oh you are right. This indeed looks right the way it is.

**Comment by xabbuh** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @JhonnyL.

**Pull request merged**